jvm-diagnostics/src/main/java/com/palantir/jvm/diagnostics/CpuSharesAccessor.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * (c) Copyright 2021 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.jvm.diagnostics;
+
+import java.util.OptionalLong;
+
+public interface CpuSharesAccessor {
+
+    /**
+     * Returns the relative-weighted CPU shares detected by the JVM. In k8s, this should be a millicore value.
+     * Returns {@link OptionalLong#empty()} if information is unavailable (and this accessor is not returned itself
+     * if cpu shares are not supported).
+     */
+    OptionalLong getCpuShares();
```

Comment:
Might be worth adding to comments that https://bugs.openjdk.org/browse/JDK-8281181 / https://bugs.openjdk.org/browse/JDK-8281571 / https://bugs.openjdk.org/browse/JDK-8279484 /  https://github.com/openjdk/jdk/commit/e07fd395bdc314867886a621ec76cf74a5f76b89 immediately deprecates `UseContainerCpuShares` in JDK 19, 11.0.17, and 17.0.5

Reply:
May be also worth mentioning https://danluu.com/cgroup-throttling/